File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.py
class Solution(object):
    def maxProduct(self, nums):
        if len(nums) < 2:
            return -1

        max_element = float('-inf')
        second_max = float('-inf')

        for num in nums:
            if num > max_element:
                second_max = max_element
                max_element = num
            elif num > second_max:
                second_max = num

        return (max_element - 1) * (second_max - 1)

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) 
    {
        if (nums.size() < 2) return -1;
        
        int f=INT_MIN,s=INT_MIN;
        
        for(auto &it: nums)
        {
            if(f<=it) {s=f;f=it;}
            else if (s < it && it < f) s = it;
        }
        
        return (f-1)*(s-1);
    }
};